autogluon/utils/tabular/ml/models/lr/lr_model.py

```diff
@@ -0,0 +1,242 @@
+        self.fit(X_train=X_train, X_test=X_test, Y_train=Y_train, Y_test=Y_test, **kwargs)
+        hpo_model_performances = {self.name: self.score(X_test, Y_test)}
+        hpo_results = {}
+        self.save()
+        hpo_models = {self.name: self.path}
+
+        return hpo_models, hpo_model_performances, hpo_results
+
+    def get_info(self):
+        # TODO: All AG-Tabular models now offer a get_info method:
+        # https://github.com/awslabs/autogluon/blob/master/autogluon/utils/tabular/ml/models/abstract/abstract_model.py#L474
+        # dict of weights?
+        return super().get_info()
+
+
+class LinearModel(AbstractLinearModel):
+    def _select_features(self, df, types_of_features, categorical_featnames, language_featnames, continuous_featnames):
+        # continuous = numeric features to rescale
+        # skewed = features to which we will apply power (ie. log / box-cox) transform before normalization
+        # onehot = features to one-hot encode (unknown categories for these features encountered at test-time are encoded as all zeros). We one-hot encode any features encountered that only have two unique values.
+        one_hot_threshold = 2 if len(language_featnames) > 0 else 10000
+        for feature in self.features:
+            feature_data = df[feature]
+            num_unique_vals = len(feature_data.unique())
+            if feature in language_featnames:
+                types_of_features['language'].append(feature)
+            elif feature in continuous_featnames:
+                if np.abs(feature_data.skew()) > self.params['proc.skew_threshold']:
+                    types_of_features['skewed'].append(feature)
+                else:
+                    types_of_features['continuous'].append(feature)
+            elif (feature in categorical_featnames) and (num_unique_vals <= one_hot_threshold):
+                types_of_features['onehot'].append(feature)
+        return types_of_features
+
+
+class LinearModelOnlyTextFeatures(AbstractLinearModel):
+    def _select_features(self, df, types_of_features, categorical_featnames, language_featnames, continuous_featnames):
+        for feature in self.features:
+            if feature in language_featnames:
+                types_of_features['language'].append(feature)
+        return types_of_features
+
+
+class LinearModelNoTextFeatures(AbstractLinearModel):
+    def _select_features(self, df, types_of_features, categorical_featnames, language_featnames, continuous_featnames):
+        # continuous = numeric features to rescale
+        # skewed = features to which we will apply power (ie. log / box-cox) transform before normalization
+        # onehot = features to one-hot encode (unknown categories for these features encountered at test-time are encoded as all zeros). We one-hot encode any features encountered that only have two unique values.
+        one_hot_threshold = 2 if len(language_featnames) > 0 else 10000
```

Comment:
Does this code even matter if there are no text features?

Reply:
Updated